compiler-api/src/main/java/com/squareup/anvil/compiler/api/AnvilContext.kt

```diff
@@ -43,6 +43,14 @@ public interface AnvilContext {
    */
   public val disableComponentMerging: Boolean
 
+  /**
+   * Track the source files for each generated file. This allows for two new behaviors:
+   * - Generated code is "invalidated" and deleted when the source file is changed or deleted.
+   * - Generated code is cached in a way that Gradle understands,
+   *   and will be restored from cache along with other build artifacts.
+   */
+  public val trackSourceFiles: Boolean
```

Comment:
Enables incremental compilation support.  This is achieved by maintaining a link between source that exists before Anvil is run and the Anvil generated files.  This allows Anvil to react to changes to sources used as input to Anvil.